connections/sim/hw/hw_sim.py

```diff
@@ -32,6 +35,20 @@ def __init__(
 
         self._paused = False
 
+    def set_pin_mode(self, pin, mode):
+        with self._lock:
+            self._pin_modes[pin] = mode
+            LOGGER.debug(f"Pin mode of pin={pin} set to={mode}")
+
+    def get_pin_mode(self, pin):
+        """
+        :param pin: Should be a test pin
+        """
+        with self._lock:
+            val = self._pin_modes[pin]
+            LOGGER.debug(f"Pin mode read from pin={pin} returned value={val}")
+            return val
+
     def digital_write(self, pin, val):
```

Comment:
In here we should assert that the pin is set as an output.

Reply:
We are now checking to see that the pin is set as an output

bd9f4ee

Reply:
This is now an assertion instead of if/else check

dea3eb7